prboom2/src/SDL/i_sound.c

```diff
@@ -502,19 +607,30 @@ static void I_UpdateSound(void *unused, Uint8 *stream, int len)
     //  as well. Thus loop those  channels.
     for ( chan = 0; chan < numChannels; chan++ )
     {
+      channel_info_t *ci = channelinfo + chan;
+
       // Check channel, if active.
-      if (channelinfo[chan].data)
+      if (ci->data)
       {
+        int s;
         // Get the raw data from the channel.
         // no filtering
         //int s = channelinfo[chan].data[0] * 0x10000 - 0x800000;
 
         // linear filtering
         // the old SRC did linear interpolation back into 8 bit, and then expanded to 16 bit.
         // this does interpolation and 8->16 at same time, allowing slightly higher quality
-        int s = ((unsigned int)channelinfo[chan].data[0] * (0x10000 - channelinfo[chan].stepremainder))
-              + ((unsigned int)channelinfo[chan].data[1] * (channelinfo[chan].stepremainder))
-              - 0x800000; // convert to signed
+        if (ci->bits == 16)
+        {
+          s = (short)(ci->data[0] | (ci->data[1] << 8)) * (255 - (ci->stepremainder >> 8))
+            + (short)(ci->data[2] | (ci->data[3] << 8)) * (ci->stepremainder >> 8);
+        }
+        else
+        {
+          s = (ci->data[0] * (0x10000 - ci->stepremainder))
```

Comment:
What is the significance of removing the cast to unsigned int from these lines?

Reply:
I just copied it from PrBoom+ without thinking. I'll restore it just in case.